robot/impl/local_robot.go

```diff
@@ -525,6 +527,10 @@ func (r *localRobot) getDependencies(
 	var needUpdate bool
 	for _, dep := range r.manager.resources.GetAllParentsOf(rName) {
 		if node, ok := r.manager.resources.Node(dep); ok {
+			// If any of the parents of the resource we're getting dependencies for
+			// has an updatedAt value that is "later" than the last value at which we
+			// ran updateWeakDependents, ensure that we run updateWeakDependents
+			// later in this method.
```

Comment:
It's unclear to me _why_ we do this check and run `updateWeakDependents` _here_. I guess if we find _some_ parent of _some_ resource has a "later" clock than the last time we ran `updateWeakDependents`, we `updateWeakDependents` later in this function. I don't understand why we can't just check the graph's logical clock with `LastUpdatedTime` (now called `LogicalClock`) and run `updateWeakDependents` if it's "later" than `lastWeakDependentsRound`. 

I'm hesitant to change this code, as it's pretty hard for me to answer the question "when do we update weak dependents?". I think it might be a lot, and, to be honest, I'm not even sure why we have all this logical clock logic _just_ to not run `updateWeakDependents` too much.

Reply:
> I don't understand why we can't just check the graph's logical clock with LastUpdatedTime (now called LogicalClock) and run updateWeakDependents if it's "later" than lastWeakDependentsRound.

I may not be understanding fully, but it seems to me that this proposal (i.e. running `updateWeakDependents` based on a comparison of `LogicalClock` and `lastWeakDependentsRound`) is logically equivalent to the current rule, but cleaner? `UpdatedAt` should always be set to a value equal to `LogicalClock` at the time of call, so unless there's some other place that `logicalClock` is getting updated somewhere else I think the proposed change is just cleaner. But this is all a bit complicated so totally reasonable to leave it as-is or save for another PR.

Reply:
> I may not be understanding fully, but it seems to me that this proposal (i.e. running updateWeakDependents based on a comparison of LogicalClock and lastWeakDependentsRound) is logically equivalent to the current rule, but cleaner?

Yep! I _think_ they'd be logically equivalent. Let me mess around locally for a bit and see if anything breaks if I move this check to the top of `updateWeakDependents` and use `CurrLogicalClockValue`.